### PR TITLE
FEAT(Day10): add color_neighbor

### PR DESCRIPTION
**Pull request by Rosein**

### CODE REVIEW

File: day_10/task_2/pipe_maze.cpp [hunk 1]
```diff
@@ -212,4 +214,101 @@ bool AttributedMaze::is_in_maze(const MazePoint& point) const
...
+    assert(is_in_maze(next_loop_point));
+    auto neighbors = find_next_neighbors(loop_point);
+    assert((next_loop_point == neighbors.first || next_loop_point == neighbors.second));
+    MazePoint second_neighbor =
+        (next_loop_point == neighbors.first) ? neighbors.second : neighbors.first;
+
+    std::vector<MazePoint> all_surrounding_tiles = {
+        MazePoint{-1, -1}, MazePoint{-1, 0}, MazePoint{-1, 1},
+        MazePoint{0, 1},   MazePoint{1, 1},  MazePoint{1, 0},
+        MazePoint{1, -1},  MazePoint{0, -1}};
```
Comment (dariannax3) [inline day_10/task_2/pipe_maze.cpp:228]: consider to move out of  the scope of this function

File: day_10/task_2/pipe_maze.cpp [hunk 2]
```diff
@@ -212,4 +214,101 @@ bool AttributedMaze::is_in_maze(const MazePoint& point) const
...
+
+    int k{};
+    while(all_surrounding_tiles.front() != next_loop_point)
+    {
+        std::vector<MazePoint>& v(all_surrounding_tiles);
+        std::rotate(v.rbegin(), v.rbegin() + 1, v.rend());
+        DEBUG_PRINT("Rotation nr" << ++k)
+        for(auto& it : all_surrounding_tiles)
+        {
+            DEBUG_PRINT("(" << it.first << ", " << it.second << ")");
```
Comment (dariannax3) [inline day_10/task_2/pipe_maze.cpp]: remove redundant debug prints

File: day_10/task_2/pipe_maze.cpp [hunk 3]
```diff
@@ -212,4 +214,101 @@ bool AttributedMaze::is_in_maze(const MazePoint& point) const
...
+    assert((next_loop_point == neighbors.first || next_loop_point == neighbors.second));
+    MazePoint second_neighbor =
+        (next_loop_point == neighbors.first) ? neighbors.second : neighbors.first;
+
+    std::vector<MazePoint> all_surrounding_tiles = {
+        MazePoint{-1, -1}, MazePoint{-1, 0}, MazePoint{-1, 1},
+        MazePoint{0, 1},   MazePoint{1, 1},  MazePoint{1, 0},
+        MazePoint{1, -1},  MazePoint{0, -1}};
+
+    std::transform(all_surrounding_tiles.begin(), all_surrounding_tiles.end(),
```
Comment (dariannax3) [inline day_10/task_2/pipe_maze.cpp:230]: global comment
consider to extract to a new function in namespace {} and add more descriptive name

File: day_10/task_2/pipe_maze.cpp [hunk 4]
```diff
@@ -212,4 +214,101 @@ bool AttributedMaze::is_in_maze(const MazePoint& point) const
...
+                   all_surrounding_tiles.begin(),
+                   [loop_point](const MazePoint& surrounding_point)
+                   {
+                       auto it = surrounding_point + loop_point;
+                       DEBUG_PRINT("(" << it.first << ", " << it.second << ")");
+                       return surrounding_point + loop_point;
+                   });
+
+    int k{};
```
Comment (dariannax3) [inline day_10/task_2/pipe_maze.cpp]: seems unnecessary when debug prints will be removed

File: day_10/task_2/pipe_maze.cpp [hunk 5]
```diff
@@ -212,4 +214,101 @@ bool AttributedMaze::is_in_maze(const MazePoint& point) const
...
+
+    for(auto& it : all_surrounding_tiles)
+    {
+        DEBUG_PRINT("(" << it.first << ", " << it.second << ")");
+    }
+
+    DEBUG_PRINT("Coloring");
+
+    auto it = all_surrounding_tiles.begin();
+    // ++it;
```
Comment (dariannax3) [inline day_10/task_2/pipe_maze.cpp]: to remove

File: day_10/task_2/tests/pipe_maze_tests.cpp [hunk 6]
```diff
@@ -133,50 +133,67 @@ INSTANTIATE_TEST_SUITE_P(IsInMazeScope,
...
+    auto color_right = std::get<3>(params);
+    auto color_down_right = std::get<4>(params);
+    auto color_down = std::get<5>(params);
+    auto color_down_left = std::get<6>(params);
+    auto color_left = std::get<7>(params);
 
+    MazePoint starting_pipe(maze.find_starting_point());
+    maze.mark_loop_tiles_in_attributed_maze();
+    const auto foundNeighbors = maze.find_next_neighbors(starting_pipe);
+    // std::cout << "BEFORE COLORING" << std::endl;
```
Comment (dariannax3) [inline day_10/task_2/tests/pipe_maze_tests.cpp]: remove

File: day_10/task_2/tests/pipe_maze_tests.cpp [hunk 7]
```diff
@@ -133,50 +133,67 @@ INSTANTIATE_TEST_SUITE_P(IsInMazeScope,
...
+    // maze.print();
```
Comment (dariannax3) [inline day_10/task_2/tests/pipe_maze_tests.cpp]: remove

Comment (Rosein) [reply day_10/task_2/tests/pipe_maze_tests.cpp]: Removed

File: day_10/task_2/tests/pipe_maze_tests.cpp (hunk 6 above)
Comment (Rosein) [reply day_10/task_2/tests/pipe_maze_tests.cpp]: Removed

File: day_10/task_2/pipe_maze.cpp (hunk 5 above)
Comment (Rosein) [reply day_10/task_2/pipe_maze.cpp]: Removed

File: day_10/task_2/pipe_maze.cpp (hunk 2 above)
Comment (Rosein) [reply day_10/task_2/pipe_maze.cpp]: Removed

File: day_10/task_2/pipe_maze.cpp (hunk 4 above)
Comment (Rosein) [reply day_10/task_2/pipe_maze.cpp]: True, removed

File: day_10/task_2/pipe_maze.cpp [hunk 8]
```diff
@@ -212,4 +213,60 @@ bool AttributedMaze::is_in_maze(const MazePoint& point) const
...
-}
\ No newline at end of file
+}
+
+MazePoint operator+(const MazePoint& lhs, const MazePoint& rhs)
+{
+    return MazePoint{lhs.first + rhs.first, lhs.second + rhs.second};
+}
+
+std::vector<MazePoint> GenerateAllSurroudningTilesStartingFromLeftTopClockWise(const MazePoint& mid_loop_point)
```
Comment (dariannax3) [inline day_10/task_2/pipe_maze.cpp]: g

**Pull request merged**